fix(css): explain flex-basis 0 v 0% (#35549)
mdn/content: files/en-us/web/css/flex-basis/index.md

  - : Any of the following units:
    - {{cssxref("&lt;length&gt;")}} sets an absolute value.

>     - {{cssxref("&lt;percentage&gt;")}} sets a percentage of the width or height of the containing block's content area. Percentage value of `flex-basis` is resolved against the flex item's flex container. If the flex container's size is indefinite, the used value for `flex-basis` is `content`.

Reviewer: ```suggestion
    - {{cssxref("&lt;percentage&gt;")}} sets a percentage of the width or height of the containing block's content area. Percentage values of `flex-basis` are resolved against the flex container. If the flex container's size is indefinite, the used value for `flex-basis` is `content`.
```

{{EmbedLiveSample('Setting_flex_item_initial_sizes', '', '360')}}

### Flex basis `0` vs `0%`

> In this example we have two flex containers with flex direction column. None of the flex containers and their flex items have height explicitly set. Flex items of the first container use flex-basie value `0` and flex items of the second container use flex-basis value `0%`. The heights of `section` elements can not exceed `200px` but their children have height `300px`.

Reviewer: ```suggestion
This example demonstrates the difference between a `flex-basis` of `0` verus a `flex-basis` of `0%` when `flex-direction` is set to `column` and the flex containers and flex items don't have a set height; while `0` is an absolute length, percentage flex-basis values resolve to [`content`](#content) values.

```

The rest is divided between the subsections

#### HTML

> 

Reviewer: ```suggestion

We include two same-structure flex containers. These which will be styled similarly, except for their `flex-basis` values. The containers each have two children: a heading `<div>` and a `<section>`. The `<section>` element has a content `<div>` child, which will not be set as a flex item but will be given a height.

```

```html
<div class="container basis-0">
  <div>heading</div>
  <section>
    flex-basis: 0;
    <div class="content"></div>
  </section>
</div>
<div class="container basis-0-percent">
  <div>heading</div>
  <section>
    flex-basis: 0%;
    <div class="content"></div>
  </section>
</div>
```

> 

Reviewer: ```suggestion

#### CSS

We style the containers as inline flex containers that will appear side by side to better enable comparing them. We set the `flex-direction` to the column. The first container's flex items have a `flex-basis` value of  `0` and the second container's flex item have a `flex-basis` value of `0%`.  Neither the flex containers nor their flex items have a height explicitly set, but the heights of `section` elements can not exceed `200px` and their children have height `300px`.

```

```css
.container {
  width: 40vw;
  padding: 1rem;
  border: 1px dashed blue;

  display: inline-flex;
  flex-direction: column;
}

section {
  border: 1px solid red;

  overflow: auto;
  min-height: 200px;
}

.content {
  background: wheat;
  height: 300px;
}

.container.basis-0 > * {
  flex-basis: 0;
}
.container.basis-0-percent > * {
  flex-basis: 0%;
}
```

> 

Reviewer: ```suggestion

#### Results

```

{{EmbedLiveSample('flex_basis_0_vs_0', '100%', '400')}}

> In the left container, with `flex-basis: 0` the section element receives initial main size of zero, and it grows till the limit `200px`. In the right container, with `flex-basis: 0%` the section element receives initial main size of `300px`. This is becaue the flex container doesn't have definite height set, so the percentage flex basis values resolve to [`content`](#content) value.

Reviewer: ```suggestion
In the first container, with `flex-basis: 0`, the `<section>` element has an initial main size of zero, and it grows to the `200px` height limit. In the second container, with `flex-basis: 0%`, the `<section>` element has an initial main size of `300px` because, as the flex container doesn't have a set height, the percentage flex-basis values  are relative to the [`content`](#content) value.
```

Author: ```suggestion
In the first container, with `flex-basis: 0`, the `<section>` element has an initial main size of zero, and it grows to the `200px` height limit. In the second container, with `flex-basis: 0%`, the `<section>` element has an initial main size of `300px` because, as the flex container doesn't have a set height, the percentage flex-basis values resolve to the [`content`](#content) value.
```
